pages/api/dpu/pubsub.ts

```diff
@@ -47,7 +47,29 @@ const pubsubHandler = async (req: NextApiRequest, res: NextApiResponse) => { //
     
     const buildStatus : CloudBuildStatus = await triggerBuildUsingGcloudApi(jsonBody.user_id, topicName);
     console.info("[pubsubHandler] build status: ", buildStatus);
-    return res.status(200).json(buildStatus);
+    if (!buildStatus.success) {
+        console.error('[pubsubHandler] Error triggering build:', buildStatus.message);
+        return res.status(500).json({ "error": buildStatus.message, success: false });
+    }
+    const projectId: string | undefined = process.env.PROJECT_ID;
+    const location: string | undefined = process.env.CLOUD_BUILD_LOCATION;
+    if (!projectId || !location) {
+        console.error('[pubsubHandler] Environment variables for projectId and location must be set');
+        res.status(400).json({"error": "env variables must be set", success: false});
```

Comment:
This should be 500, right? The problem occurred from our side - didn't set the environment variables. 400 is for bad request.